modules/backend/widgets/Filter.php

```diff
@@ -561,6 +562,12 @@ public function addScopes(array $scopes)
         foreach ($scopes as $name => $config) {
             $scopeObj = $this->makeFilterScope($name, $config);
 
+            // Check if user has permissions to show this filter
+            $permissions = array_get($config, 'permissions');
+            if (!empty($permissions) && !BackendAuth::getUser()->hasAccess($permissions, false)) {
+                continue;
+            }
```

Comment:
@Samuell1 can you move this to be the first check of the loop? There's no need to make the $scopeObj if we're not going to use it.